File: src/main/java/io/github/dqualizer/dqlang/types/adapter/symbolics/generic/SymbolicDurationType.java
package io.github.dqualizer.dqlang.types.adapter.symbolics.generic;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.ToString;

@Getter
@ToString
public class SymbolicDurationType<T> {

    private T slow;
    private T fast;
    @JsonProperty("very_fast")
    private T veryFast;
}

File: src/main/kotlin/io/github/dqualizer/dqlang/types/adapter/symbolics/generic/SymbolicDurationType.kt
package io.github.dqualizer.dqlang.types.adapter.symbolics.generic

import com.fasterxml.jackson.annotation.JsonProperty

data class SymbolicDurationType<T>(
  val slow: T,

  val fast: T,

  @JsonProperty("very_fast")
  val veryFast: T,
)
